### PR TITLE
AB#150 Added Left Menu Print

### PR DESCRIPTION
**Pull request by CurtisUnderwood**
### Description

List of proposed changes:

- Added Print Button in left menu of task page (not visible on mobile)
- Added custom print styles to only display expanded accordion
- Added custom class to include href at the end of each link in the accordions

### What to test for/How to test
- Try Printing Task page and confirm all correct elements are displayed

### Additional Notes

### CODE REVIEW

File: frontend/src/components/NestedAccordion.tsx [hunk 1]
```diff
@@ -20,7 +20,15 @@ const NestedAccordion: React.FC<NestedAccordionProps> = ({
   tasks = [],
 }) => {
   return (
-    <Accordion className="mb-4">
+    <Accordion className="mb-4"
+    sx={{
+      "@media print": {
+        ".mui-style-bz4dnt-MuiCollapse-root": {
```
Comment (Fbasham) [inline frontend/src/components/NestedAccordion.tsx:26]: This goes back to one of Greg's comments on another PR.  It should be possible to exactly target this element without the dynamic [bs4dnt]...albeit kind of annoying/trial and error.

Comment (CurtisUnderwood) [reply frontend/src/components/NestedAccordion.tsx]: Fixed!

**Comment by Fbasham** [comment]
image


Testing in Firefox.  There appears to be some overflow clipping.  I don't even know if it's possible to prevent on the vertical. I'd imagine the horizontal clipping can be handled though.

**Comment by CurtisUnderwood** [comment]
> image
> 
> Testing in Firefox. There appears to be some overflow clipping. I don't even know if it's possible to prevent on the vertical. I'd imagine the horizontal clipping can be handled though.
> 
> [possible help](https://stackoverflow.com/questions/16221412/css-media-print-causes-part-of-the-page-to-be-cut-off-during-printing-in-ie)

image

Believe this is resolved by changing the print display to block.

**Comment by sebastien-comeau** [comment]
This should be a different Page or Page Component. It would simplify the logic and give more flexibility. Just my 2 cents.

File: frontend/src/components/Footer.tsx [hunk 2]
```diff
@@ -73,8 +74,10 @@ const Footer: FC<FooterProps> = ({
   menuLinks,
   links,
 }: FooterProps) => {
+  const { pathname } = useRouter()
+  const isTasksPage = pathname.includes('/tasks');
```
Comment (sebastien-comeau) [inline frontend/src/components/Footer.tsx:78]: This should be handle in `Layout` component.

Add a `hideHeader` and `hideFooter` props to Layout component with value of `undefined | "never" | "always" | "print"`. Then you implemtent the logic in `Layout` component then the page should pass the props.

File: frontend/src/components/Layout.tsx [hunk 3]
```diff
@@ -11,106 +11,114 @@ export interface LayoutProps {
...
-        skipToMainText={t('header.skip-to-main')}
-        gocLink={t('header.goc-link')}
-        breadcrumbItems={breadcrumbItems}
-        hideChecklist={hideChecklist}
+      {hideHeader !== 'always' && 
+      <Header 
+        skipToMainText={t('header.skip-to-main')} 
+        gocLink={t('header.goc-link')} 
+        breadcrumbItems={breadcrumbItems} 
+        className={`${hideHeader == 'print' && "print:hidden"}`}
```
Comment (sebastien-comeau) [inline frontend/src/components/Layout.tsx:28]:
```suggestion
        className={hideHeader == 'print' ? "print:hidden" : undefined}
```

File: frontend/src/components/Layout.tsx [hunk 4]
```diff
@@ -11,106 +11,114 @@ export interface LayoutProps {
...
-          },
-          {
-            link: t('footer.menu.man-url'),
-            linkText: t('footer.menu.man'),
-          },
-        ]}
-      />
+      {hideFooter !== 'always' && 
+        <Footer
+          className={`${hideFooter == 'print' && "print:hidden"}`}
```
Comment (sebastien-comeau) [inline frontend/src/components/Layout.tsx:38]:
```suggestion
          className={hideFooter == 'print' ? "print:hidden" : undefined}
```

File: frontend/src/pages/home.tsx [hunk 5]
```diff
@@ -70,7 +70,12 @@ const Home: FC = () => {
   }
 
   return (
-    <Layout breadcrumbItems={[]} contained={false}>
+    <Layout
+    breadcrumbItems={[]}
+    contained={false}
+    hideFooter='never'
+    hideHeader='never'
```
Comment (sebastien-comeau) [inline frontend/src/pages/home.tsx:77]: Remove it if it's `never` (undefined equals 'never') and assign Layout's defaultProps:
```tsx
Layout.defaultProps: { 
  hideFooter: 'never'
  hideHeader: 'never'
}

File: frontend/src/styles/globals.css [hunk 6]
```diff
@@ -25,6 +25,12 @@
   --tw-content: '';
 }
 
+/*@media print {
+  a:after {
+    content: " (" attr(href) ")";
+  }
+}*/
+
```
Comment (sebastien-comeau) [inline frontend/src/styles/globals.css:33]: Remove that comment.

Comment (CurtisUnderwood) [reply frontend/src/styles/globals.css]: Fixed

File: frontend/src/pages/home.tsx (hunk 5 above)
Comment (CurtisUnderwood) [reply frontend/src/pages/home.tsx]: Updated

File: frontend/src/components/Layout.tsx (hunk 4 above)
Comment (CurtisUnderwood) [reply frontend/src/components/Layout.tsx]: Fixed

File: frontend/src/components/Layout.tsx (hunk 3 above)
Comment (CurtisUnderwood) [reply frontend/src/components/Layout.tsx]: Fixed

**Review by sebastien-comeau — APPROVED** [review]

**Pull request merged**